### PR TITLE
[Python] Display progress bar by default in an interactive environment

### PR DESCRIPTION
**Pull request by Tishj**
This PR addresses a part of #5407 

Namely this addresses:
- > Enable progress bars by default in an interactive environment.
- > Output HTML when executed in Jupyter Notebook (probably more customizations to come)

Note however that the second point is not entirely fixed by this PR, this merely addresses the overlap of point two with point one.

Only progress bars are handled differently in Jupyter for now.

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! Looks great. Making the progress bar display a virtual class is a great idea.

**Comment by Mytherin** [comment]
Can we not (temporarily) grab the GIL during the call to `Update`?

**Comment by Tishj** [comment]
@Mytherin It seems to work!
Tested locally that both regular interactive mode and jupyter progress bars work with these changes

File: tools/pythonpkg/src/jupyter/jupyter_progress_bar_display.cpp [hunk 1]
```diff
@@ -20,11 +25,20 @@ JupyterProgressBar::JupyterProgressBar() {
 	display_attr(progress_bar);
 }
 
-void JupyterProgressBar::Update(double progress) {
+JupyterProgressBarDisplay::JupyterProgressBarDisplay() : ProgressBarDisplay() {
+	// Empty, we need the GIL to initialize, which we don't have here
```
Comment (Tishj) [inline tools/pythonpkg/src/jupyter/jupyter_progress_bar_display.cpp:29]: We could also grab the GIL here, but we really don't need the initialization to happen here, I figured reducing GIL grabbing would be preferable

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**